compiler/generator/jax/jax_code_container.cpp

```diff
@@ -0,0 +1,411 @@
+    } else if (gGlobal->gVectorSwitch) {
+        throw faustexception("ERROR : Vector not supported for JAX\n");
+    } else {
+        container = new JAXScalarCodeContainer(name, numInputs, numOutputs, dst, kInt);
+    }
+
+    return container;
+}
+
+inline string flattenJSONforPython(const string& src)
+{
+    string dst;
+    for (size_t i = 0; i < src.size(); i++) {
+        switch (src[i]) {
+            case '"':
+                dst += "\\\"";
+                break;
+            case '\\':
+                dst += "/";
+                break;
+            case '\'':
+                dst += "'";
+                break;
+            default:
+                dst += src[i];
+                break;
+        }
+    }
+    return dst;
+}
+
+void JAXCodeContainer::produceClass()
+{
+    int n = 0;
+    
+    // Print header
+    *fOut << "\"\"\"" << endl
+          << "Code generated with Faust version " << FAUSTVERSION << endl;
+    *fOut << "Compilation options: ";
+    stringstream stream;
+    gGlobal->printCompilationOptions(stream);
+    *fOut << stream.str();
+    tab(n, *fOut);
+    *fOut << "\"\"\"";
+    tab(n, *fOut);
+    
+    tab(n, *fOut);
+    *fOut << "import jax";
+    tab(n, *fOut);
+    *fOut << "import jax.numpy as jnp";
+    tab(n, *fOut);
+    *fOut << "from flax import linen as nn";
+    tab(n, *fOut);
+    *fOut << "import librosa";
+    tab(n, *fOut);
+    *fOut << "import json";
+    tab(n, *fOut);
+    *fOut << "import re";
+    tab(n, *fOut);
+
+    tab(n, *fOut);
+    tab(n, *fOut);
+    *fOut << "def FAUSTFLOAT(x):";
+    tab(n+1, *fOut);
+    *fOut << "return x";
+    tab(n, *fOut);
+        
+    // Merge sub containers
+    mergeSubContainers();
+
+    // Functions
+    tab(n, *fOut);
+    gGlobal->gJAXVisitor->Tab(n);
+       
+    tab(n, *fOut);
+    *fOut << "class " << fKlassName << "(nn.Module):";
+    tab(n + 1, *fOut);
+
+    // Fields
+    gGlobal->gJAXVisitor->Tab(n + 1);
+
+    tab(n + 1, *fOut);
+    *fOut << "sample_rate: int";
+    tab(n + 2, *fOut);
+
+    // Get sample rate method
+    tab(n, *fOut);
+    gGlobal->gJAXVisitor->Tab(n);
+
+    tab(n+1, *fOut);
+    produceInfoFunctions(n+1, "", "self", false, false, gGlobal->gJAXVisitor);
+    
+    *fOut << "def classInit(self, state):";
+    {
+        tab(n + 2, *fOut);
+        *fOut << "# global declarations:";
+        JAXInitFieldsVisitor initializer(fOut, n + 2);
+        generateDeclarations(&initializer);
+        // Generate global variables initialisation
+        for (const auto& it : fGlobalDeclarationInstructions->fCode) {
+            if (dynamic_cast<DeclareVarInst*>(it)) {
+                it->accept(&initializer);
+            }
+        }
+        tab(n + 2, *fOut);
+        *fOut << "# inline subcontainers:";
+        tab(n + 2, *fOut);
+        gGlobal->gJAXVisitor->Tab(n + 2);
+        inlineSubcontainersFunCalls(fStaticInitInstructions)->accept(gGlobal->gJAXVisitor);
```

Comment:
I'm testing `import("stdfaust.lib"); process = os.osc(440.);`

At this point I get code like
```python
# inline subcontainers:
l0_re0 = 0
while (l0 < 2):
	state["iVec0"]  = state["iVec0"].at[l0].set(0) 
	l0 = (l0 + 1) 
```
but `l0_re0` should be `l0` and I haven't figured out how to fix it.